### DISCUSSION TITLE
Migrating from 5.7 to 6.1.3

### DISCUSSION THREAD

**Discussion by polif7** [panva/openid-client#723] [Q&A]
Hi,
I have a question regarding the new version of this library.
I'm using Angular SSR and I want to implement BFF for authorization - [datatracker](https://datatracker.ietf.org/doc/html/draft-ietf-oauth-browser-based-apps#name-backend-for-frontend-bff)

Right now I have a problem with finding solution how to properly set and use this library. All docs and tutorials are for the old methods. Like here - [passport](https://www.passportjs.org/packages/openid-client/).

I found some POC [repo](https://github.com/ErikWegner/bffssrng/blob/master/ngbff/server/common/auth.ts) to catch some general idea of this whole flow, but I'm missing this crucial steps:
- Issuer.discover
- myIssuer.Client
- passport.use('oidc', new Strategy(?))

Could someone help me? How it should look like in new v6 way of things?

**Comment by panva** [comment]
Hello @polif7 

Examples and documentation are linked from the README in the very repository you're asking the question in 🤷

**Reply by polif7** [reply]
Thank you for your answer!
Yes with this [file](https://github.com/panva/openid-client/blob/main/examples/passport.ts) I can get my third point.
Caould you point me what now exacly stands for Issuer.discover + myIssuer.Client?

**Reply by panva** [reply]
L30

**Reply by polif7** [reply]
Hi, I've run on another problem.
Our auth server is localized, so its hosted on different domains. For example url is something like login.us and issuer is always the same - login.nu.
When I'm using this `await client.discovery(authUrl, clientId, {client_secret: clientSecret});` I have an error 
`discovered metadata issuer does not match the expected issuer`
`OAUTH_JSON_ATTRIBUTE_COMPARISON_FAILED`

As I understand the first argument stands for server url and also issuer.
How can I pass this valid "login.nu" issuer? and also use localized server url?

**Reply by panva** [reply]
As per the specification, your issuer identifier input must match the discovered one. For localization purposes you may want to use the `ui_locales` authorization endpoint parameter instead.

If that's something you absolutely can't achieve, just pass the discovery document's endpoint URL (with the .well-known pathname) instead of the issuer identifier to discovery.
